components/library/objfw/Makefile

```diff
@@ -0,0 +1,59 @@
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright (c) 2023, Jonathan Schleifer. All rights reserved.
+#
+
+BUILD_BITS = 64
+USE_PARALLEL_BUILD = yes
+
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME = objfw
+COMPONENT_VERSION = 1.0.3
+COMPONENT_SUMMARY = A portable, lightweight framework for the Objective-C language
+COMPONENT_SRC = $(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE = $(COMPONENT_SRC).tar.gz
+COMPONENT_ARCHIVE_HASH = sha256:1c81d7d03578b2d9084fc5d8722d4eaa4bdc2f3f09ce41231e7ceab8212fae17
+COMPONENT_ARCHIVE_URL = https://objfw.nil.im/downloads/$(COMPONENT_ARCHIVE)
+COMPONENT_PROJECT_URL = https://objfw.nil.im/
+COMPONENT_FMRI = library/objfw
+COMPONENT_CLASSIFICATION = Development/Objective C
+COMPONENT_LICENSE = QPL OR GPLv3 OR GPLv2
+COMPONENT_LICENSE_FILE = objfw.license
+
+include $(WS_MAKE_RULES)/common.mk
+
+# The build system does not support building out of tree.
+COMPONENT_PRE_CONFIGURE_ACTION = ($(CLONEY) $(SOURCE_DIR) $(@D))
+
+CONFIGURE_OPTIONS += --disable-rpath
+CONFIGURE_OPTIONS += --with-tls=gnutls
+CONFIGURE_OPTIONS += OBJC=clang
```

Comment:
I think you should use full `clang` path here.  The full `clang` path is `$(CLANG_PREFIX)/bin/clang`.

Reply:
That results in `/usr/clang/15.0/bin/clang: No such file or directory`

Reply:
Ah, `CLANG_VERSION = 17` helps.

Reply:
Install `developer/clang-15`, or wait until #13947 is merged.

Reply:
Why? It works as expected with setting `CLANG_VERSION = 17`.

Reply:
Because such line will need to be either updated or removed in future.  It is always better to create `Makefile` that does not need such babysitting.

Reply:
This seems like a bad idea, as clang-17 is added as a dependency, but clang 15 would be used to actually compile. Even if I would change the dependency to clang-15, it would go out of sync as soon as the default is updated without anyone noticing.

The other issue is that Clang is often broken on non-mainstream OSes, especially when it involves exceptions. So manually setting it to a known working version would be better. In fact, this discussion just made me realize that clang *is* generating broken code on OpenIndiana for AMD64. I didn't notice because the tests aren't run by default. I guess I'll have to try different clang versions and see if one works for AMD64.

Reply:
You should run tests, ofc.  And provide test results.

Yes, some tests for clang fails (55 failures for clang-17, 45 failures for clang-15).  So maybe the problem with broken code on OpenIndiana for AMD64 is well known (and silently accepted).

I do not understand why people integrate software with known test failures and without any attempt to address such failures.

Reply:
The test suite runs fine for 32 bit. The used clang version doesn't matter, so the issue is likely in gcc_s. Is there a way to use clang with a different GCC version?

For now, I'll just make it 32 bit only.

Reply:
> This seems like a bad idea, as clang-17 is added as a dependency, but clang 15 would be used to actually compile. Even if I would change the dependency to clang-15, it would go out of sync as soon as the default is updated without anyone noticing.

You are right.  You should not need to add default clang as dependency.  This was not the case, so I created #13948.

Reply:
I guess in this case the pkg5 file should also no longer reference clang?

Reply:
Fixed the issue with unwinding on 64 bit. It was, as expected, libgcc_s. Interestingly, -lgcc_s and just linking with the full path of the .so leads to different results. But this works now.

Reply:
Removed explicit dependency on clang.